src/data_provider/src/packages/appxWindowsWrapper.h

```diff
@@ -0,0 +1,454 @@
+            std::string name;
+            constexpr auto INVALID_NAME_APP { "@{" };
+            auto findCacheName
+            {
+                [&appName](const std::string folder)
+                {
+                    return folder.find(appName) != std::string::npos;
+                }
+            };
+
+            const auto folder = std::find_if(m_cacheReg.begin(), m_cacheReg.end(), findCacheName);
+            // Search for the name of the folder containing the application name
+
+            if (folder != m_cacheReg.end())
+            {
+                registry = *folder;
+            }
+
+            if (!registry.empty())
+            {
+                name = searchKeyOnSubRegistries(m_userId + "\\" + CACHE_NAME_REGISTRY + "\\" + registry, nameKey);
+            }
+
+            const auto index { name.find(INVALID_NAME_APP) };
+            return (index != std::string::npos && index == 0) ? "" : name;
+        }
+
+        /*
+         * @brief Search the key on the registry and sub-registries.
+         *
+         * @param[in] path Start path where you want to search.
+         * @param[in] key Name key to look for
+         *
+         * @return Return the value of the key.
+         */
+        const std::string searchKeyOnSubRegistries(const std::string& path, const std::string& key)
+        {
+            std::string value;
+            const Utils::Registry registry(m_key, path);
+
+            if (!registry.string(key, value))
+            {
+                for (const auto& folder : Utils::Registry(m_key, path).enumerate())
+                {
+                    const std::string tempPath { path + "\\" + folder };
+                    value = searchKeyOnSubRegistries(tempPath, key);
+
+                    if (!value.empty())
+                    {
+                        break;
+                    }
+                }
+            }
+
+            return value;
+        }
+
+        /*
+         * @brief Search the publisher name in a registry.
+         *
+         * @param[in] registry Registry object used to obtain the information.
+         *
+         * @return Returns the publisher name found.
+         */
+        const std::string searchPublisher(const Utils::Registry& registry)
+        {
+            std::string publisher;
+
+            for (const auto& value : registry.enumerateValueKey())
+            {
+                std::string data;
+                std::string vendorRegistry;
+
+                registry.string(value, vendorRegistry);
+                const Utils::Registry pubRegistry(m_key, m_userId  + "\\" + APPLICATION_VENDOR_REGISTRY + "\\" + vendorRegistry + "\\Application");
+
+                if (pubRegistry.string("ApplicationCompany", data))
+                {
+                    const auto index { data.find("@{") };
+
+                    if (index != 0)
+                    {
+                        publisher = data;
+                        break;
+                    }
```

Comment:
Utils::startsWith could be useful here?
Add global constexpr auto PREFIX_LOCALIZATION {"@{"};
```suggestion
                    if (!Utils::startsWith(data, PREFIX_LOCALIZATION))
                    {
                        publisher = data;
                        break;
                    }
```